### DISCUSSION TITLE
Need help enabling Asciidoc detection

### DISCUSSION THREAD

**Discussion by LouvAndTech** [github-linguist/linguist#7100] [Q&A] [answered]
Hello,

I need your help understanding my mistake here.
I have a repository for documentation writing that use mostly Asciidoc. 
I notice that some repository on GitHub have asciidoc as part of their statistics, like [this one](https://github.com/starknet-io/starknet-docs).
I tried using the overwrite with the .gitattributes file like so:
```
➜ git:(main) ✗ cat .gitattributes
# Detect .asciidoc, .adoc and .yml files
*.adoc linguist-detectable
*.asciidoc linguist-detectable
*.yml linguist-detectable#
```
But after commit and push asciidoc still is missing from the stats... 
```
➜ git:(main) ✗ github-linguist .
74.06%  4463       Python
19.96%  1203       YAML
5.97%   360        CSS
```

YAML on the other hand appeared without an issue.
I tried to make a small change in both an .adoc and a .asciidoc file but the issue persisted.

Hope someone can help me.

**Comment by lildude** [comment, marked as answer]
You've not provided a link to your repo so I can't see things in action, but given your explanation that this is for documentation purposes, it's possible your files are in a directory Linguist considers documentation and thus ignores:

https://github.com/github-linguist/linguist/blob/f164d13fa618023ecf2d8f2ed9a6ce5fae731346/lib/linguist/documentation.yml#L10-L19

In this case you will need to also add the `-linguist-documentation` [override](https://github.com/github-linguist/linguist/blob/main/docs/overrides.md#documentation).

If this isn't the case, we'd really need to see the repo or a mock that illustrates the problem if your repo is private.

**Reply by LouvAndTech** [reply]
That was exactly my problem! 
Added : `doc/** -linguist-documentation` to my `.gitattribute` solved the issue.
Thank you verry much!
